internal/provider/resources/variable.go

```diff
@@ -136,9 +156,93 @@ func (r *VariableResource) Schema(_ context.Context, _ resource.SchemaRequest, r
 	}
 }
 
+// UpgradeState adds upgraders to the VariableResource.
+// This is needed when a resource schema change is made (eg. an attribute type).
+// The key/index in the return object is the source version (eg. 0 -> current).
+// The target version is the one defined in Schema.Version above
+// https://developer.hashicorp.com/terraform/plugin/framework/resources/state-upgrade
+func (r *VariableResource) UpgradeState(_ context.Context) map[int64]resource.StateUpgrader {
+	defaultEmptyTagList, _ := basetypes.NewListValue(types.StringType, []attr.Value{})
+
+	return map[int64]resource.StateUpgrader{
+		// State upgrade implementation from prior (0) => current (Schema.Version)
+		0: {
+			// PriorSchema allows the framework to populate the req.State argument
+			// for easier data handling when migrating existing state resources
+			// to a new schema.
+			PriorSchema: &schema.Schema{
```

Comment:
I'm on mobile so I may be wrong, but would it make sense to change this schema definition to a variable and refer to it that way instead of redefining it? Not a blocker but a passing thought

Reply:
good call. we're just "updating" one of the attributes (aka the one that changed and requires this schema migration)

i'll stash the current schema in a var and then map those keys onto the PriorSchema while modifying the one attribute we need to match the older schema